```
fn gcd(mut n: u64, mut m: u64) -> u64 {

    assert!(n != 0 && m != 0);

    // 余数不为0

    while m != 0 {

        // 如果m小于n，交换

        if m < n {

            let t = m;

            m = n;

            n = t;

        }

        // 需要保留两者中小的数

        // 计算出两者的余数，作为下一轮计算的参数

        // 不需要保留大的数，大的数就等于余数

        m = m % n;

    }

    n

}
```
`assert!(n != 0 && m != 0);`
```
   Compiling gcd v0.1.0 (/home/l/projects/rust_book/rust-programming-design/01/gcd)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.20s
     Running `target/debug/gcd`

thread 'main' panicked at 01/gcd/src/main.rs:6:5:
assertion failed: n != 0 && m != 0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
如果非真，则终止本程序并提供一条有帮助的信息，其中包括导致本次检查失败的源代码位置。这种突然的终止在 Rust 中称为 panic。与可以跳过断言的 C 和 C++ 不同，Rust 总是会检查这些断言，而不管程序是如何编译的。还有一个 debug_assert!宏，在编译发布版程序时会跳过其断言以提高速度。
[debug_assert！](../../../../Rust/复用概念/断言/debug_assert！.md)

[1.0 自动推断类型](../../../../Rust/复用概念/类型/1.0%20自动推断类型.md)